segmentation/utils/spatial_analysis_utils.py

```diff
@@ -44,46 +75,78 @@ def compute_close_cell_num(patient_data_markers, label_idx, thresh_vec,
     index [0, 1]).
 
     Args:
-        patient_data_markers: cell expression data of markers
-            for the specific point
-        label_idx: column of cell labels
-        thresh_vec: matrix of thresholds column for markers
         dist_mat: cells x cells matrix with the euclidian
             distance between centers of corresponding cells
-        marker_num: number of markers in expresion data
         dist_lim: threshold for spatial enrichment distance proximity
+        num: number of markers or cell phenotypes, based on analysis
+        analysis_type: type of analysis, either Cell Label or Threshold
+        point: current point doing analysis on
+        all_patient_data: data for all patients in expression matrix
+        data_markers: data of only column markers for Channel Analysis
+        pheno_codes: list all the cell phenotypes in Cell Label Analysis
+        thresh_vec: matrix of thresholds column for markers
 
     Returns:
         close_num: marker x marker matrix with counts for cells
             positive for corresponding markers
         marker1_num: list of number of cell labels for marker 1
         marker2_num: list of number of cell labels for marker 2"""
+    # Initialize variables
+    label_idx = []
+    patient_data_markers = pd.DataFrame()
+
+    # Assign column names for subsetting (patient IDs, cell labels, cell phenotypes)
+    patient_idx = "SampleID"
+    cell_label_idx = "cellLabelInImage"
+    flowsom_idx = "FlowSOM_ID"
+
+    # Subset data based on analysis type
+    if analysis_type == "Cell Label":
+        # Subset matrix to only include the columns with the patient label, cell label, and cell phenotype
+        all_patient_data_pheno = all_patient_data[[patient_idx, cell_label_idx, flowsom_idx]]
+        # Subsetting expression matrix to only include patients with correct label
+        patient_ids = all_patient_data_pheno.iloc[:, 0] == point
```

Comment:
Sorry, that wasn't clear on my part. The patient_ids subsetting should still occur in the main function, since we're looping over each patient in the for loop. None of these functions should have to deal with patient-specific indexing at all, that will be handled by the loop itself